schema.sql

```diff
@@ -1,12 +1,66 @@
 /* Database schema to keep the structure of entire database. */
 
+DROP TABLE animals;
+
+/* Build 1.1 */
+
 CREATE TABLE animals (
-    id INT,
+    id INT GENERATED BY DEFAULT AS IDENTITY,
     name varchar(100),
     date_of_birth DATE,
     escape_attempts INT,
     neutered BOOLEAN,
-    weight_kg DECIMAL
+    weight_kg DECIMAL,
+    PRIMARY KEY(id)
 );
 
--- DROP TABLE animals;
+/* Build 1.2 */
+ALTER TABLE animals
+ADD COLUMN species varchar;
+
+BEGIN;
+    UPDATE animals
+    SET species = NULL;
+COMMIT;
+
+BEGIN;
+    UPDATE animals
+    SET species = 'Unspecified';
+ROLLBACK;
+
+BEGIN;
+    UPDATE animals
+    SET species = 'digimon'
+    WHERE name LIKE '%mon';
+COMMIT;
+
+BEGIN;
+    UPDATE animals
+    SET species = 'pokemon'
+    WHERE species IS NULL;
+COMMIT;
+
+BEGIN;
+    DELETE FROM animals;
+ROLLBACK;
+
+UPDATE animals
```

Comment:
* Kindly move your transactions queries to the queries.sql file as Everything that creates or changes the table's structure should be saved in the schema.sql file.

Reply:
Done, thank you 👍🏾